### DISCUSSION TITLE
Choose PV and RBD image name

### DISCUSSION THREAD

**Discussion by M-Gregoire** [rook/rook#14043] [Q&A] [answered]
Hello,

When creating a `PersistentVolumeClaim` using a rook `StorageClass`, the `PersistentVolume` and associated Ceph RBD image are created automatically with a randomly generated name like `pvc-<uuid>`.

It it possible to specify the name  of the PV and the Image inside Ceph to browse them more easily?

Thanks

**Comment by Madhu-1** [comment, marked as answer]
Hey @M-Gregoire this is not possible for dynamic pv provisioning

**Reply by M-Gregoire** [reply]
Thanks a lot, that's settles it then!
I guess I should still be able to create a `PersistentVolume` manually to set the name right? If so, is there any documentation available by any chance? I couldn't find anything in the docs or in the examples.

**Reply by Madhu-1** [reply]
Yes possible here is the doc https://github.com/ceph/ceph-csi/blob/devel/docs/static-pvc.md, this wont support resize/clone etc.

You have other options as below

* you can set custom prefix per storageclass https://github.com/ceph/ceph-csi/blob/devel/examples/rbd/storageclass.yaml#L116
* If you check the pv -oyaml you already have all the required details like rbd/cephfs image name, PVC name etc
* We also set PVC related metadata in rbd image/subvolume as well. https://github.com/ceph/ceph-csi/pull/3272#issue-1321710967 and [rbd image-meta get](https://github.com/ceph/ceph-csi/pull/2898/files)

**Reply by M-Gregoire** [reply]
Awesome! Thanks a lot for your comprehensive summary, that answers it perfectly!
